commit dfdfc2beb0dd7e3a067d2eeacb4623cb48e77658
Author: Sven Eckelmann <sven@narfation.org>
Date:   Tue Jan 26 17:11:13 2016 +0100

    mac80211: Parse legacy and HT rate in injected frames
    
    Drivers/devices without their own rate control algorithm can get the
    information what rates they should use from either the radiotap header of
    injected frames or from the rate control algorithm. But the parsing of the
    legacy rate information from the radiotap header was removed in commit
    e6a9854b05c1 ("mac80211/drivers: rewrite the rate control API").
    
    The removal of this feature heavily reduced the usefulness of frame
    injection when wanting to simulate specific transmission behavior. Having
    rate parsing together with MCS rates and retry support allows a fine
    grained selection of the tx behavior of injected frames for these kind of
    tests.
    
    Signed-off-by: Sven Eckelmann <sven@narfation.org>
    Cc: Simon Wunderlich <sw@simonwunderlich.de>
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>

diff --git a/Documentation/networking/mac80211-injection.txt b/Documentation/networking/mac80211-injection.txt
index 3a930072b161..ec8f934c2eb2 100644
--- a/Documentation/networking/mac80211-injection.txt
+++ b/Documentation/networking/mac80211-injection.txt
@@ -28,6 +28,23 @@ radiotap headers and used to control injection:
    IEEE80211_RADIOTAP_F_TX_NOACK: frame should be sent without waiting for
 				  an ACK even if it is a unicast frame
 
+ * IEEE80211_RADIOTAP_RATE
+
+   legacy rate for the transmission (only for devices without own rate control)
+
+ * IEEE80211_RADIOTAP_MCS
+
+   HT rate for the transmission (only for devices without own rate control).
+   Also some flags are parsed
+
+   IEEE80211_TX_RC_SHORT_GI: use short guard interval
+   IEEE80211_TX_RC_40_MHZ_WIDTH: send in HT40 mode
+
+ * IEEE80211_RADIOTAP_DATA_RETRIES
+
+   number of retries when either IEEE80211_RADIOTAP_RATE or
+   IEEE80211_RADIOTAP_MCS was used
+
 The injection code can also skip all other currently defined radiotap fields
 facilitating replay of captured radiotap headers directly.